### DISCUSSION TITLE
Customize Oauth2 client

### DISCUSSION THREAD

**Discussion by dbarrientostk** [pysnippet/fastapi-oauth2#35] [Q&A]
How I can access to the refresh token?
I want to customize the oauth methods and create new ones.

**Comment by ArtyomVancyan** [comment]
Hi @dbarrientostk, refresh tokens are handled by [social-core](https://github.com/python-social-auth/social-core), and you don't need to do additional work for it. You can have the [access token](https://docs.pysnippet.org/fastapi-oauth2/references/tutorials#access-token) that your provider issued for you and use for API calls.

**Reply by dbarrientostk** [reply]
I have created a new "backend" using social core, therefore I need to handle the refresh token a little different, also I need to revoke the token in the logout method.
Could you please provide an example?
@ArtyomVancyan 
<img width="786" alt="Screenshot 2024-03-12 at 09 48 30" src="https://github.com/pysnippet/fastapi-oauth2/assets/156249225/afbfd9d7-bb2c-42dd-90eb-dc5446495ac2">

Config.py
<img width="710" alt="Screenshot 2024-03-12 at 10 09 22" src="https://github.com/pysnippet/fastapi-oauth2/assets/156249225/57a068bd-5bc4-42e8-9b26-169e03526a89">

Here is the request.auth:

<img width="947" alt="Screenshot 2024-03-12 at 10 05 25" src="https://github.com/pysnippet/fastapi-oauth2/assets/156249225/2996e693-4f05-4bc4-a90c-19f65489ae33">

**Reply by ArtyomVancyan** [reply]
Anyway, the refresh token handling must be done by the social-auth, and if you look at the [backend implementation](https://python-social-auth.readthedocs.io/en/latest/backends/implementation.html) documentation, you will find `REFRESH_TOKEN_URL` that need to be set. If it still does not solve your issue, try to override the base class methods to achieve your needs.

**Reply by dbarrientostk** [reply]
@ArtyomVancyan  which base clase do I have to modify?

**Reply by ArtyomVancyan** [reply]
In the documentation link above, you'll find three examples: Oauth2, Oauth, OpenId. Three of them have base classes that you should inherit to create your own authentication method. I think for you it will be the `BaseOAuth2` class.

```python
from social_core.backends.oauth import BaseOAuth2
```

**Reply by ArtyomVancyan** [reply]
@dbarrientostk, have you fixed your problem?
